### PR TITLE
Add fmt formatter for SymInt

### PR DESCRIPTION
**Pull request by ezyang**
Stack from [ghstack](https://github.com/ezyang/ghstack) (oldest at bottom):
* __->__ #113784

Signed-off-by: Edward Z. Yang <ezyang@meta.com>

### CODE REVIEW

File: c10/CMakeLists.txt [hunk 1]
```diff
@@ -88,7 +88,7 @@ endif()
 if(C10_USE_GLOG)
     target_link_libraries(c10 PUBLIC glog::glog)
 endif()
-target_link_libraries(c10 PRIVATE fmt::fmt-header-only)
+target_link_libraries(c10 PUBLIC fmt::fmt-header-only)
```
Comment (albanD) [inline c10/CMakeLists.txt:91]: Do we really have to?

Comment (albanD) [inline c10/CMakeLists.txt:91]: Do we really have to? We have fmt formatting for other things and we never needed that right?

Comment (ezyang) [reply c10/CMakeLists.txt:91]: I cannot `#include <fmt/format.h>` without this. It is very awkward.

There are calls to fmt/format.h functions in c10, but no custom formatting (which is a template specialization). I'm not sure if this is at all resolvable without putting the template specialization where SymInt is defined.

Comment (albanD) [reply c10/CMakeLists.txt:91]: This leads to compilation errors I'm afraid :/

Comment (ezyang) [reply c10/CMakeLists.txt:91]: yeah I give up lol

**Pull request closed without merging**